### PR TITLE
Fix controller runtime package update from v0.15.1 to v0.16.1

### PR DESCRIPTION
**Pull request by msherif1234**
**- What this PR does and why is it needed**

**- Special notes for reviewers**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: cmd/daemon/daemon.go [hunk 1]
```diff
@@ -15,24 +15,27 @@ package main
 
 import (
 	"flag"
+	"k8s.io/client-go/rest"
```
Comment (martinkennelly) [inline cmd/daemon/daemon.go:18]: format

File: cmd/daemon/daemon.go [hunk 2]
```diff
@@ -15,24 +15,27 @@ package main
...
 	// Import all Kubernetes client auth plugins (e.g. Azure, GCP, OIDC, etc.)
 	// to ensure that exec-entrypoint and run can make use of them.
 	_ "k8s.io/client-go/plugin/pkg/client/auth"
 
+	ingressnodefwiov1alpha1 "github.com/openshift/ingress-node-firewall/api/v1alpha1"
+	ingressnodefwv1alpha1 "github.com/openshift/ingress-node-firewall/api/v1alpha1"
+	"github.com/openshift/ingress-node-firewall/controllers"
+	"github.com/openshift/ingress-node-firewall/pkg/metrics"
+	"github.com/openshift/ingress-node-firewall/pkg/version"
 	"k8s.io/apimachinery/pkg/runtime"
```
Comment (martinkennelly) [inline cmd/daemon/daemon.go:31]: [stdlib]

[local]

[remote/k8s]

I think thats how we usually structure it.

File: main.go [hunk 3]
```diff
@@ -86,8 +87,7 @@ func main() {
 
 	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
 		Scheme:                 scheme,
-		MetricsBindAddress:     metricsAddr,
-		Port:                   9443,
```
Comment (martinkennelly) [inline main.go:90]: where do we define the port now?

File: pkg/webhook/webhook_suite_test.go [hunk 4]
```diff
@@ -22,6 +22,7 @@ import (
 	"fmt"
 	"net"
 	"path/filepath"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
```
Comment (martinkennelly) [inline pkg/webhook/webhook_suite_test.go:25]: format

File: main.go (hunk 3 above)
Comment (msherif1234) [reply main.go:90]: from what I have seen its seems to be bundled with metricAddr I wish if they called `BindSocket` instead

**Comment by msherif1234** [comment]
/retest
cluster didn't come up

File: main.go (hunk 3 above)
Comment (msherif1234) [reply main.go:90]: I need to investigate further this port is used by webhook server 
```
	// Port is the port that the webhook server serves at.
	// It is used to set webhook.Server.Port if WebhookServer is not set.
	//
	// Deprecated: Use WebhookServer instead. A WebhookServer can be created via webhook.NewServer.
	Port int
```

Comment (msherif1234) [reply main.go:90]: fixed

**Pull request merged**